### PR TITLE
FPGAs changes and fixup, ethernet, pcie, constraints

### PR DESCRIPTION
**Pull request by CyrilKoe**
Changes:
* Re-wired properly the JTAG chain on island disable
* Added the safety + spatz config
* Corrected pulp cluster CDC constraints
* Modified device tree structure to have a new PCIe top level dts
* Modified PCIe ranges in block design
* Added reconfigurable MAC address and ethernet boot image path
* Added CVA6-sdk as a subrepo as in Cheshire
* Modified CI accordingly ([please see PR](https://iis-git.ee.ethz.ch/carfield/carfield-nonfree/-/merge_requests/50)) (auto select available board + remote boot possible)

Status:
* Ready, [boots over ethernet](https://iis-git.ee.ethz.ch/github-mirror/carfield/-/jobs/1237290#L7) and [pulp cluster implements](https://iis-git.ee.ethz.ch/github-mirror/carfield/-/pipelines/86456.

### CODE REVIEW

File: sw/deps/cva6-sdk
Comment (yvantor) [inline sw/deps/cva6-sdk:1]: Would it be possible to use HTTPS also for buildroot dependency in CVA6-SDK? Something similar to what was done [here](https://github.com/pulp-platform/cva6-sdk/commit/175b88bae504757f93e59d02d437bdfdd886dd2c) (cherry-picking should be east, I can also open a PR if you want)

**Review by yvantor — COMMENTED** [review]
Just one comment on the CVA6-SDK, overall I did not spot anything proud of notice. One question, I see you have added a DTS for booting on VCU118 and also made some of the scripts more "agnostic" towards the kind of board. Do you think it would make sense to also merge in this branch the changes to support the VCU118?

File: sw/deps/cva6-sdk
Comment (CyrilKoe) [reply sw/deps/cva6-sdk:1]: Done

**Comment by CyrilKoe** [comment]
> Just one comment on the CVA6-SDK, overall I did not spot anything proud of notice. One question, I see you have added a DTS for booting on VCU118 and also made some of the scripts more "agnostic" towards the kind of board. Do you think it would make sense to also merge in this branch the changes to support the VCU118?

I removed it, it will be part of the next PR

**Comment by CyrilKoe** [comment]
@yvantor or @alex96295  this is now ready for review

**Comment by CyrilKoe** [comment]
CI failed due to limited disk space, restarted it this morning

**Comment by CyrilKoe** [comment]
What do you think @yvantor @alex96295  ?

**Review by yvantor — APPROVED** [review]
Hey! I confirm my previous comment, I do not have particular concerns. Approved!

**Comment by alex96295** [comment]
Thanks! I will review it next week

File: carfield.mk [hunk 1]
```diff
@@ -44,7 +44,7 @@ include $(CAR_ROOT)/bender-safed.mk
 ######################
 
 CAR_NONFREE_REMOTE ?= git@iis-git.ee.ethz.ch:carfield/carfield-nonfree.git
-CAR_NONFREE_COMMIT ?= 54ce7e49
+CAR_NONFREE_COMMIT ?= 59e531343b31c4787c1ff72ab0e18365ea97fbb9
```
Comment (alex96295) [inline carfield.mk:47]: Please use short hash

**Review by alex96295 — APPROVED** [review]
Just a very small comment, for the rest LGTM; thanks for the work

File: carfield.mk (hunk 1 above)
Comment (CyrilKoe) [reply carfield.mk]: Done

**Pull request merged**